adaptive simulcast change log update (#1623)
twilio/twilio-video.js: CHANGELOG.md

New Features
------------

> This release introduces new beta feature **Adaptive Simulcast**. This opt-in feature can be enabled by setting `preferredVideoCodecs="auto"` in connect options. When joining a group room with this feature enabled SDK will use VP8 simulcast, and will enable/disable simulcast layers dynamically, thus improving bandwidth and CPU usage. It works best when used along with `Client Track Switch Off Controls` and `Video Content Preferences`. These two flags allows SFU to determine which simulcast layers are needed, thus allowing it to disable the layers not needed on publisher side.

Reviewer: ```suggestion
This release introduces new beta feature **Adaptive Simulcast**. This opt-in feature can be enabled by setting `preferredVideoCodecs="auto"` in connect options. When joining a group room with this feature enabled the SDK will use VP8 simulcast, and will enable/disable simulcast layers dynamically, thus improving bandwidth and CPU usage. It works best when used along with `Client Track Switch Off Controls` and `Video Content Preferences`. These two flags allows the SFU to determine which simulcast layers are needed, thus allowing it to disable the layers not needed on publisher side.
```

> This release introduces new beta feature **Adaptive Simulcast**. This opt-in feature can be enabled by setting `preferredVideoCodecs="auto"` in connect options. When joining a group room with this feature enabled the SDK will use VP8 simulcast, and will enable/disable simulcast layers dynamically, thus improving bandwidth and CPU usage. It works best when used along with `Client Track Switch Off Controls` and `Video Content Preferences`. These two flags allows the SFU to determine which simulcast layers are needed, thus allowing it to disable the layers not needed on publisher side. The beta currently does not support setting a max bitrate for your simulcast layers.

Reviewer: ```suggestion
This release introduces a new beta feature **Adaptive Simulcast**. This opt-in feature can be enabled by setting `preferredVideoCodecs="auto"` in connect options. When joining a group room with this feature enabled the SDK will use VP8 simulcast, and will enable/disable simulcast layers dynamically, thus improving bandwidth and CPU usage. It works best when used along with `Client Track Switch Off Controls` and `Video Content Preferences`. These two flags allows the SFU to determine which simulcast layers are needed, thus allowing it to disable the layers not needed on publisher side. The beta currently does not support setting a max bitrate for your simulcast layers.
```

> This release introduces a new beta feature **Adaptive Simulcast**. This opt-in feature can be enabled by setting `preferredVideoCodecs="auto"` in connect options. When joining a group room with this feature enabled the SDK will use VP8 simulcast, and will enable/disable simulcast layers dynamically, thus improving bandwidth and CPU usage. It works best when used along with `Client Track Switch Off Controls` and `Video Content Preferences`. These two flags allows the SFU to determine which simulcast layers are needed, thus allowing it to disable the layers not needed on publisher side. The beta currently does not support setting a max bitrate for your simulcast layers.

Reviewer 2: ```suggestion
This release introduces a new beta feature **Adaptive Simulcast**. This opt-in feature can be enabled by setting `preferredVideoCodecs="auto"` in ConnectOptions. When joining a group room with this feature enabled, the SDK will use VP8 simulcast, and will enable/disable simulcast layers dynamically, thus improving bandwidth and CPU usage. It works best when used along with `Client Track Switch Off Controls` and `Video Content Preferences`. These two flags allows the SFU to determine which simulcast layers are needed, thus allowing it to disable the layers not needed on publisher side. The beta currently does not support setting a max bitrate for your simulcast layers.
```

  });
  ```

> Additional notes:

Reviewer: ```suggestion
Known Limitations:
```

> - This feature is incompatible with connect option `maxVideoBitrate`, and will result in an error at connect time if specified along with the new option.

Reviewer: I think this should be moved up higher so this section can be focused on known limitations. This is more instructions for use. I would add the following known limitation:

"The beta currently does not support setting a max bitrate for your simulcast layers."

- This feature is not supported on some configurations. In following cases specifying `preferredVideoCodecs="auto"` will revert back to using unicast.
  - Firefox,
  - When room is configured to use H264 codec
  - Peer-to-Peer rooms.

> - When room is being recorded, SFU will not disable layers, as they need to be recorded.

Reviewer: ```suggestion
- When the room or participant publishing is being recorded, the SFU will not disable any simulcast layers.
```

  });
  ```

Known Limitations
-----------------

> - This feature is not supported on some configurations. In following cases specifying `preferredVideoCodecs="auto"` will revert back to using unicast.

Reviewer: ```suggestion
- This feature is not supported on some configurations. In the following cases specifying `preferredVideoCodecs="auto"` will revert back to using unicast.
```

  - When publisher is using firefox

>   - When room is configured to use H264 codec

Reviewer: ```suggestion
  - When room is configured to use H.264 codec
```

> - This feature is not supported on some configurations. In the following cases specifying `preferredVideoCodecs="auto"` will revert back to using unicast.
>   - When publisher is using firefox
>   - When room is configured to use H.264 codec
>   - Peer-to-Peer rooms
> - When the room or participant publishing is being recorded, the SFU will not disable any simulcast layers.

Reviewer 2: ```suggestion
- Specifying `preferredVideoCodecs="auto"` will revert to unicast in the following cases:
  - The publisher is using Firefox.
  - The publisher has preferred the H264 codec.
  - The Room is configured to support only the H264 codec.
  - Peer-to-Peer Rooms
- When the Room is being recorded, the SFU will not disable any simulcast layers of the publisher's VideoTrack.
```
